### ISSUE TITLE
Does not work with `gulp-istanbul`

### ISSUE THREAD

**Issue by yuanqing** [yuanqing/gulp-tape#3]
This doesn't work:

``` js
var tape = require('gulp-tape');
var istanbul = require('gulp-istanbul');

gulp.task('test', function(cb) {
  gulp.src('js/**/*.js')
    .pipe(istanbul())
    .pipe(istanbul.hookRequire())
    .on('finish', function() {
      gulp.src('test/*.js')
        .on('error', cb)
        .on('end', cb)
        .pipe(tape())
        .pipe(istanbul.writeReports({
          reporters: ['lcov', 'text']
        }));
    });
});
```

**Comment by zoubin** [comment]
I use istanbul in [this way](https://github.com/zoubin/es6-pkg/blob/master/gulpfile.babel.js#L47).
It seems to work.

**Comment by zoubin** [comment]
I tried your example:

lib/main.js

``` javascript
exports.abs = function abs(val) {
  if (val > 0) {
    return val;
  }
  if (val === 0) {
    return 0;
  }
  return -val;
}

exports.halve = function halve(i) {
  return i >> 1;
}
```

test/main.js

``` javascript
var test = require('tape');
var lib = require('../lib/main.js');
var abs = lib.abs;
var halve = lib.halve;

test('abs', function(t) {
  t.equal(abs(1), 1, 'positive');
  t.equal(abs(-1), 1, 'negative');
  t.end();
});

test('halve', function(t) {
  t.equal(halve(3), 1, 'odd');
  t.equal(halve(4), 2, 'even');
  t.end();
});
```

gulpfile.js

``` javascript
var gulp = require('gulp');

gulp.task('test', require('callback-sequence')(
  instrument, test, report
));

function test() {
  var tape = require('gulp-tape');
  return gulp.src('test/*.js')
    .pipe(tape());
}

function instrument() {
  var istanbul = require('gulp-istanbul');
  return gulp.src('lib/*.js')
    .pipe(istanbul())
    .pipe(istanbul.hookRequire())
}

function report() {
  var istanbul = require('gulp-istanbul');
  return gulp.src('test/*.js', { read: false })
  .pipe(istanbul.writeReports({
    reporters: ['lcov', 'text'],
  }));
}
```

and the output of `gulp test` :

```
⌘ gulp test
[09:23:14] Using gulpfile ~/tmp/gulpfile.js
[09:23:14] Starting 'test'...
TAP version 13
# abs
ok 1 positive
ok 2 negative
# halve
ok 3 odd
ok 4 even
----------|----------|----------|----------|----------|----------------|
File      |  % Stmts | % Branch |  % Funcs |  % Lines |Uncovered Lines |
----------|----------|----------|----------|----------|----------------|
 lib/     |     87.5 |       75 |      100 |     87.5 |                |
  main.js |     87.5 |       75 |      100 |     87.5 |              6 |
----------|----------|----------|----------|----------|----------------|
All files |     87.5 |       75 |      100 |     87.5 |                |
----------|----------|----------|----------|----------|----------------|

[09:23:14] Finished 'test' after 382 ms

1..4
# tests 4
# pass  4

# ok
```

**Comment by yuanqing** [comment]
Fixed by #4
